spec/Section 6 -- Execution.md

```diff
@@ -479,17 +479,28 @@ The depth-first-search order of the field groups produced by {CollectFields()}
 is maintained through execution, ensuring that fields appear in the executed
 response in a stable and predictable order.
 
+When {CollectFields()} is used during validation (see for example the
+[single root field](#sec-Single-root-field) subscription operation validation
+rule), the runtime value for {variableValues} will not be available - in this
+case we set {variableValues} to {null} and forbid the use of the `@skip` and
+`@include` directives. During execution, {variableValues} will always be
+non-null.
+
```

Comment:
This is pretty confusing and feels like a hack. I'd rather see validation self-contained and not require the edge case within execution that's potentially not used by execution.